qt/widgets/plotting/inc/MantidQtWidgets/Plotting/PreviewPlot.h

```diff
@@ -122,8 +120,8 @@ public slots:
   void createLayout();
   void createActions();
 
-  void onWorkspaceRemoved(Mantid::API::WorkspacePreDeleteNotification_ptr nf);
-  void onWorkspaceReplaced(Mantid::API::WorkspaceBeforeReplaceNotification_ptr nf);
+  void replaceHandle(const std::string &wsName, const Workspace_sptr &ws) override;
+  void deleteHandle(const std::string &wsName, const Workspace_sptr &ws) override;
```

Comment:
what do you think about naming these `replaceWorkspaceHandle` and `deleteWorkspaceHandle`?

Reply:
I think `replaceWorkspaceHandle` and `deleteWorkspaceHandle` sound like better names! However, these are overriding the methods in the `AnalysisDataServiceObserver` which is used extensively in Mantid. I would need to rename in every place this is used, so I'm not sure it would be a quick change